lightning-payjoin/src/error.rs

```diff
@@ -0,0 +1,123 @@
+use lightning::util::errors::APIError;
+
+#[derive(Debug)]
+pub enum Error {
+	Internal(InternalError),
+	Payjoin(payjoin::Error),
+}
+
+#[derive(Debug)]
+pub enum InternalError {
+	TokioJoinError(tokio::task::JoinError),
+	FailedBuildingResponse(hyper::http::Error),
+	WalletOperationFailed,
+	OnchainTxSigningFailed,
+	FundingTxGenerationFailed(String),
+	NodeFailed,
+}
+
+impl From<tokio::task::JoinError> for InternalError {
+	fn from(value: tokio::task::JoinError) -> Self {
+		Self::TokioJoinError(value)
+	}
+}
+
+impl From<hyper::http::Error> for InternalError {
+	fn from(value: hyper::http::Error) -> Self {
+		Self::FailedBuildingResponse(value)
+	}
+}
+
+impl From<hyper::http::Error> for Error {
+	fn from(value: hyper::http::Error) -> Self {
+		Self::Internal(InternalError::FailedBuildingResponse(value))
+	}
+}
+
+impl From<tokio::task::JoinError> for Error {
+	fn from(value: tokio::task::JoinError) -> Self {
+		Self::Internal(InternalError::TokioJoinError(value))
+	}
+}
+
+impl From<InternalError> for Error {
+	fn from(value: InternalError) -> Self {
+		Self::Internal(value)
+	}
+}
+
+impl From<payjoin::Error> for Error {
+	fn from(value: payjoin::Error) -> Self {
+		Self::Payjoin(value)
+	}
+}
+
+impl From<Error> for payjoin::Error {
+	fn from(value: Error) -> Self {
+		match value {
+			Error::Payjoin(e) => e,
+			_ => unreachable!(),
+		}
+	}
+}
+
+impl std::fmt::Display for InternalError {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+		match *self {
+			Self::TokioJoinError(ref e) => write!(f, "Tokio join error: {}", e),
+			Self::FailedBuildingResponse(ref e) => write!(f, "Failed building response: {}", e),
+			Self::WalletOperationFailed => write!(f, "Wallet operation failed"),
+			Self::OnchainTxSigningFailed => write!(f, "Onchain transaction signing failed"),
+			Self::FundingTxGenerationFailed(ref e) => {
+				write!(f, "Failed to generate funding transaction: {}", e)
+			},
+			Self::NodeFailed => write!(f, "Node failed"),
+		}
+	}
+}
+
+impl std::fmt::Display for Error {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+		match *self {
+			Self::Internal(ref e) => write!(f, "Internal lightning_payjoin error: {}", e),
+			Self::Payjoin(ref e) => write!(f, "Payjoin error: {}", e),
+		}
+	}
+}
+
+// Maybe this should live in LDK-Node instead?
```

Comment:
Oh sorry what I meant here is implement it to the ldk node `Error` rather than for `lightning-payjoin::Error` so we dont add here stuff that are specific for ldk node handling